packages/remote-feature-flag-controller/src/remote-feature-flag-controller.test.ts

```diff
@@ -0,0 +1,303 @@
+    disabled: options.disabled,
+  });
+}
+
+describe('RemoteFeatureFlagController', () => {
+  let clientConfigApiService: AbstractClientConfigApiService;
+
+  describe('constructor', () => {
+    it('should initialize with default state', () => {
+      const controller = createController();
+
+      expect(controller.state).toStrictEqual({
+        remoteFeatureFlags: [],
+        cacheTimestamp: 0,
+      });
+    });
+
+    it('should initialize controller with default state if the disabled parameter is provided', () => {
+      const controller = createController({ disabled: true });
+
+      expect(controller.state).toStrictEqual({
+        remoteFeatureFlags: [],
+        cacheTimestamp: 0,
+      });
+    });
+
+    it('should initialize controller with custom state', () => {
+      const customState = {
+        remoteFeatureFlags: mockFlagsTwo,
+        cacheTimestamp: 123456789,
+      };
+
+      const controller = createController({ state: customState });
+
+      expect(controller.state).toStrictEqual(customState);
+    });
+  });
+
+  describe('getRemoteFeatureFlags', () => {
+    it('should return empty object when controller is disabled', async () => {
+      clientConfigApiService = buildClientConfigApiService();
+      const controller = createController({
+        clientConfigApiService,
+        disabled: true,
+      });
+
+      const remoteFeatureFlags = await controller.getRemoteFeatureFlags();
+
+      expect(remoteFeatureFlags).toStrictEqual([]);
+      expect(
+        clientConfigApiService.fetchRemoteFeatureFlags,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('should make network request to fetch when cache is not expired', async () => {
+      clientConfigApiService = buildClientConfigApiService();
+      const controller = createController({ clientConfigApiService });
+      const flags = await controller.getRemoteFeatureFlags();
+
+      expect(flags).toStrictEqual(mockFlags);
+    });
+
+    it('should make network request to fetch when cache is expired', async () => {
+      clientConfigApiService = buildClientConfigApiService({
+        cacheTimestamp: Date.now() - 10000,
+      });
+      const controller = createController({
+        clientConfigApiService,
+        disabled: false,
+      });
+
+      // Mock different response
+      jest
+        .spyOn(clientConfigApiService, 'fetchRemoteFeatureFlags')
+        .mockImplementation(async () => ({
+          error: false,
+          message: 'Success',
+          statusCode: '200',
+          statusText: 'OK',
+          cachedData: mockFlagsTwo,
+          cacheTimestamp: Date.now(),
+        }));
+
+      const remoteFeatureFlags = await controller.getRemoteFeatureFlags();
+      expect(remoteFeatureFlags).toStrictEqual(mockFlagsTwo);
+      expect(
+        clientConfigApiService.fetchRemoteFeatureFlags,
+      ).toHaveBeenCalledTimes(1);
+    });
+
+    it('should not affect existing cache when toggling disabled state', async () => {
```

Comment:
Since this is primarily testing the `disable` functionality, I'll move to a different test and simplify it , addressed in [614997a0e55603083347c47f9e04d74a2055db50](https://github.com/MetaMask/core/commit/41bb909507cd6ac641f10d06ae1e99ef8de240ce)
```
  describe('disable', () => {
    it('should preserve cached flags but return empty array when disabled', async () => {
 ```